### ISSUE TITLE
Implement clowder processing of local files

### ISSUE THREAD

**Issue by dlebauer** [terraref/computing-pipeline#63]
Currently clowder works on upload. need to be able to point to a local file. 

This is described in and will be implemented by https://opensource.ncsa.illinois.edu/jira/browse/CATS-324

**Comment by robkooper** [comment]
See https://opensource.ncsa.illinois.edu/jira/browse/CATS-324

**Comment by max-zilla** [comment]
There is currently a pull request for this - just needs to be reviewed.

**Comment by dlebauer** [comment]
Have you added this to the documentation? Is this something that I should assign to @robkooper to merge?

And .. is it enabled on the http://141.142.208.144/clowder/collections instance?

**Comment by max-zilla** [comment]
@dlebauer I'll add it to the documentation - not specifically described yet.

It is not currently enabled on that instance - that instance is running the Clowder master branch which is even further behind than the develop branch at 209.122, but the team hopes to have a "v1.0" for the master branch in the near future. Once my [pull request](https://opensource.ncsa.illinois.edu/bitbucket/projects/CATS/repos/clowder/pull-requests/743/overview) is merged into clowder and that instance is updated, the local file stuff should be enabled.

Be aware though that there is more work to be done here. We don't currently have the 'listening' daemon that will monitor the incoming globus file directory and add files to Clowder after transfer. Unfortunately globus API does not have a simple way to query whether transfer is complete or fire an event on completion - @robkooper and I can go into more detail tomorrow. My pull request allows Clowder to point to local files, but we still need the piece before that that actually tells Clowder when to do so.

**Comment by dlebauer** [comment]
@robkooper I believe this is waiting on your to review this pull request: https://opensource.ncsa.illinois.edu/bitbucket/projects/CATS/repos/clowder/pull-requests/743/overview

@max-zilla I don't see how to upload local files in the instructions, which I think is separate from the 'listening daemon'. Presumably this is referring to #62 'globus file status checker'?

**Comment by max-zilla** [comment]
@dlebauer yes, the listener refers to #62.

As far as the local file upload instructions, I can write something up but it depends on the use case you're thinking of here. For typical users, they would just load files into clowder the usual way (sending via UI or API). These local file instructions would, I imagine, be intended for someone deploying a new instance of Clowder who wanted to set up a similar Globus pipeline or something.

Local files in this context refer to files located on the same server as the Clowder database - workflow would be:
1. Update clowder's config file with a whitelist of folders on server that are accepted as local file sources
2. Transmit file(s) out to a whitelisted folder on the server, e.g. via FTP
3. Send a POST to the Clowder API that points to that file using the server local path
4. File gets an entry in clowder without needing to copy the bytes into the Mongo database or elsewhere - your files will remain in the folder you uploaded them to with same filenames, etc. 

Does that fit with what you're thinking of?

**Comment by dlebauer** [comment]
Specific use case is https://github.com/terraref/computing-pipeline/issues/41#issuecomment-194440338

I downloaded a bunch of files that will be run through a genomics pipeline (could be done independently of Clowder, but seems useful to have an extractor run the remote pipeline). Perhaps we won't need to do this on local files; in this case I can move them elsewhere and then upload them to the API.

**Comment by robkooper** [comment]
Looking at pull request, there are some issues with it that needs to be cleaned up before it can be merged.

**Comment by max-zilla** [comment]
This was merged and is currently the means by which incoming Globus files are added to Clowder in the gantry -> Clowder pipeline - #83 

Closing issue. Adding a note in the documentation issue 5 to get this documented as well:
https://github.com/terraref/documentation/issues/5
